[#566] Add meta-schema.json and test using it.

kindly opened the pull request:
related #566

kindly committed:
[#566] Add meta-schema.json and test using it.

jpmckinney commented on standard/schema/meta-schema.json:
```diff
@@ -0,0 +1,190 @@
+        "anyOf": { "$ref": "#/definitions/schemaArray" },
+        "oneOf": { "$ref": "#/definitions/schemaArray" },
+        "not": { "$ref": "#" },
+        "$ref": {
```
For `$ref`, draft-06 has:

```
        "$ref": {
            "type": "string",
            "format": "uri-reference"
        },
```

though we'd use `"format": "uri"` for draft-04.

kindly replied:
I think the issue here is that internal $ref i.e '#/definitions/Awards' would not pass "uri" validataion. Which I think is why they added the new format.

jpmckinney replied:
Aha, makes sense.

jpmckinney commented on standard/schema/meta-schema.json:
```diff
@@ -0,0 +1,190 @@
+            "type": "boolean",
+            "default": false
+        }
+
```
Extra line.

jpmckinney reviewed (commented):
Made a few comments. It might be good for easier change-tracking to first commit the draft-04 schema, and then commit our changes, so that we can easily see the divergence when reviewing history.

kindly committed:
[#556] Make metaschema patch.

Add script make_metaschema.py to make meta-schema.json from patch and
draft-04.  Also add test to make sure this is run.

jpmckinney commented on standard/schema/tests/test_validate_schema.py:
```diff
@@ -31,14 +31,15 @@ def object_pairs_hook(pairs):
 
 
 DRAFT_SCHEMA_PATH = path(__file__).parent / 'json-schema-draft-4.json'
```
This constant's value needs to be updated because the file moved in another commit, but if we aren't using the constant anymore, we can just remove it.

kindly committed:
[#566] Remove variabile to draft04 from test.
